pyomo/contrib/mindtpy/feasibility_pump.py

```diff
@@ -39,288 +37,63 @@ def fp_converged(solve_data, config, discrete_only=True):
-            solve_data.results.solver.status = SolverStatus.error
-            return
-        else:
-            raise ValueError(
-                'MindtPy unable to handle NLP subproblem termination '
-                'condition of {}'.format(fp_nlp_result.solver.termination_condition))
-        # Call the NLP post-solve callback
-        config.call_after_subproblem_solve(fp_nlp, solve_data)
-        solve_data.fp_iter += 1
-    solve_data.mip.MindtPy_utils.del_component('fp_mip_obj')
-
-    if config.fp_main_norm == 'L1':
-        solve_data.mip.MindtPy_utils.del_component('L1_obj')
-    elif config.fp_main_norm == 'L_infinity':
-        solve_data.mip.MindtPy_utils.del_component(
-            'L_infinity_obj')
-
-    # deactivate the improving_objective_cut
-    solve_data.mip.MindtPy_utils.cuts.del_component(
-        'improving_objective_cut')
-    if not config.fp_transfercuts:
-        for c in solve_data.mip.MindtPy_utils.cuts.oa_cuts:
-            c.deactivate()
-        for c in solve_data.mip.MindtPy_utils.cuts.no_good_cuts:
-            c.deactivate()
-    if config.fp_projcuts:
-        solve_data.working_model.MindtPy_utils.cuts.del_component(
-            'fp_orthogonality_cuts')
-
-
-def add_orthogonality_cuts(solve_data, config):
+def add_orthogonality_cuts(working_model, mip_model, config):
     """Add orthogonality cuts.
 
     This function adds orthogonality cuts to avoid cycling when the independence constraint qualification is not satisfied.
 
     Parameters
     ----------
-    solve_data : MindtPySolveData
-        Data container that holds solve-instance data.
+    working_model : Pyomo model
+        The working model(original model).
+    mip_model : Pyomo model
+        The mip model.
     config : ConfigBlock
         The specific configurations for MindtPy.
     """
-    mip_integer_vars = solve_data.mip.MindtPy_utils.discrete_variable_list
-    nlp_integer_vars = solve_data.working_model.MindtPy_utils.discrete_variable_list
+    mip_integer_vars = mip_model.MindtPy_utils.discrete_variable_list
+    nlp_integer_vars = working_model.MindtPy_utils.discrete_variable_list
     orthogonality_cut = sum((nlp_v.value-mip_v.value)*(mip_v-nlp_v.value)
                             for mip_v, nlp_v in zip(mip_integer_vars, nlp_integer_vars)) >= 0
-    solve_data.mip.MindtPy_utils.cuts.fp_orthogonality_cuts.add(
+    mip_model.MindtPy_utils.cuts.fp_orthogonality_cuts.add(
         orthogonality_cut)
     if config.fp_projcuts:
         orthogonality_cut = sum((nlp_v.value-mip_v.value)*(nlp_v-nlp_v.value)
                                 for mip_v, nlp_v in zip(mip_integer_vars, nlp_integer_vars)) >= 0
-        solve_data.working_model.MindtPy_utils.cuts.fp_orthogonality_cuts.add(
+        working_model.MindtPy_utils.cuts.fp_orthogonality_cuts.add(
             orthogonality_cut)
 
 
-def generate_norm_constraint(fp_nlp, solve_data, config):
+def generate_norm_constraint(fp_nlp, mip, config):
```

Comment:
Just to be consistent with the functions above, let's call all the models in the parameters as `..._model`